src/subcommand/miner/metrics.rs

```diff
@@ -7,114 +7,126 @@ pub(crate) struct Metrics {
 }
 
 impl Metrics {
-    pub fn new() -> Self {
+    pub(crate) fn new() -> Self {
         Self {
             hashes: AtomicU64::new(0),
             shares: AtomicU64::new(0),
             started: Instant::now(),
         }
     }
```

Comment:
Responsive hashrate display step 2: Initialize the hashrate field in `new()`
```suggestion
    pub(crate) fn new() -> Self {
        Self {
            hashes: AtomicU64::new(0),
            shares: AtomicU64::new(0),
            started: Instant::now(),
            hashrate: std::sync::Mutex::new(DecayingAverage::new(Duration::from_secs(10))),
        }
    }
```